### PR TITLE
fix: handle the case that the name characters is too long

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
 handle the case that the name characters is too long

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=328288&iterationID=1321&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     handle the case that the name characters is too long          |
| 🇨🇳 中文    |        处理字符数过长的情况      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/dop/utils/pipeline.yml.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/erda-project/erda/pkg/parser/pipelineyml"
+)
+
+func MakeProjectPipelineName(pipelineYml string, fileName string) string {
+	yml, err := pipelineyml.GetNameByPipelineYml(pipelineYml)
+	if err == nil && yml != "" {
+		return yml
+	}
+	name := filepath.Base(fileName)
+	if utf8.RuneCountInString(name) <= 30 {
```
Comment (sfwn) [inline internal/apps/dop/utils/pipeline.yml.go:30]: `30` should be an function arg.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/utils/pipeline.yml.go (hunk 1 above)
Comment (littlejiancc) [reply internal/apps/dop/utils/pipeline.yml.go]: done

**Review by sfwn — DISMISSED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**